### PR TITLE
Find users nearby

### PR DESCRIPTION
**Pull request by michal-kk**
Stupid solution, I'm surprised it works. Should be replaced by something more real. It should be possible to get those in one mongo query/aggregation.

### CODE REVIEW

File: src/main/java/com/example/helpgiver/controller/UserController.java [hunk 1]
```diff
@@ -133,4 +133,29 @@
                 new CollectionModel<>(userEntities,
                         linkTo(methodOn(UserController.class).getUserGeo(x, y, distanceKm)).withSelfRel()));
     }
+
+    // TODO other method like this return <GeoResult<User>>, but do they need to?
+    @GetMapping("helpRequest/{id}/nearbyUsers")
```
Comment (shumanski) [inline src/main/java/com/example/helpgiver/controller/UserController.java:138]: /potentialHelpers ?
nearbyUsers is confusing

File: src/main/java/com/example/helpgiver/controller/UserController.java [hunk 2]
```diff
@@ -133,4 +133,29 @@
...
+    public ResponseEntity<CollectionModel<EntityModel<User>>> getUsersNearHelpRequest(@PathVariable String id) {
+        Optional<HelpRequest> optionalHelpRequest = helpRequestRepository.findById(id);
+
+        if (optionalHelpRequest.isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+
+        // TODO it should be possible to replace this monstrosity with a mongo query:
+        Collection<EntityModel<User>> usersNearby = userRepository.findAll().stream()
```
Comment (shumanski) [inline src/main/java/com/example/helpgiver/controller/UserController.java:147]: Yes, it is very ugly :)

You can initially search by max possible radius, I think now it is 10Km

File: src/main/java/com/example/helpgiver/controller/UserController.java (hunk 1 above)
Comment (michal-kk) [reply src/main/java/com/example/helpgiver/controller/UserController.java:138]: I wanted to use `users` beause it names a "type", but I can change

File: src/main/java/com/example/helpgiver/controller/UserController.java (hunk 2 above)
Comment (michal-kk) [reply src/main/java/com/example/helpgiver/controller/UserController.java:147]: oh, good. I was thinking about this, but wasn't sure if there is a max.

File: src/main/java/com/example/helpgiver/controller/UserController.java (hunk 1 above)
Comment (shumanski) [reply src/main/java/com/example/helpgiver/controller/UserController.java:138]: nearbyUsers is way too confusing. Not sure about exact name, but it shall be changed.

Comment (michal-kk) [reply src/main/java/com/example/helpgiver/controller/UserController.java:138]: `potentiallyHelpingUsers`? to long?

Comment (shumanski) [reply src/main/java/com/example/helpgiver/controller/UserController.java:138]: OK... for now.

**Pull request merged**